File: app/src/main/java/rubik/shifttest/presentation/greeting/GreetingViewModel.java
package rubik.shifttest.presentation.greeting;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.rubik.shifttest.domain.models.UserRegisterCredential;
import com.rubik.shifttest.domain.usecases.GetUserRegisterCredentialUseCase;

public class GreetingViewModel extends ViewModel {

    private final GetUserRegisterCredentialUseCase getUserRegisterCredentialUseCase;
    private final MutableLiveData<UserRegisterCredential> userRegisterCredentialLive;

    public GreetingViewModel(GetUserRegisterCredentialUseCase getUserRegisterCredentialUseCase) {
        this.getUserRegisterCredentialUseCase = getUserRegisterCredentialUseCase;
        userRegisterCredentialLive = new MutableLiveData<>();
    }

    public void getUserRegisterCredential() {
        userRegisterCredentialLive.setValue(getUserRegisterCredentialUseCase.execute());
    }

    public LiveData<UserRegisterCredential> getUserRegisterCredentialLive() {
        return userRegisterCredentialLive;
    }
}

File: feature/user/greeting/src/main/kotlin/ru/rubik/greeting/presentation/GreetingViewModel.kt
package ru.rubik.greeting.presentation

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch
import ru.rubik.user.domain.repository.UserRepository

class GreetingViewModel(
	private val repository: UserRepository,
) : ViewModel() {

	private val _uiState = MutableStateFlow<GreetingState>(GreetingState.Button)
	val uiState: StateFlow<GreetingState> = _uiState

	fun getUserInfo() {
		viewModelScope.launch {
			_uiState.value = GreetingState.Dialog(repository.getUser().firstName)
		}
	}
}